src/styleguide/renderer/react/preview.tsx

```diff
@@ -1,321 +0,0 @@
-import { AssetProperty } from '../../../store/styleguide/property/asset-property';
-import * as Chokidar from 'chokidar';
-import { ErrorMessage } from './error-message';
-import { HighlightArea } from '../highlight-area';
-import * as MobX from 'mobx';
-import { observer } from 'mobx-react';
-import { Page } from '../../../store/page/page';
-import { PageElement } from '../../../store/page/page-element';
-import * as PathUtils from 'path';
-import { Pattern } from '../../../store/styleguide/pattern';
-import { PatternComponent } from './pattern-component';
-import { Placeholder } from './placeholder';
-import * as React from 'react';
-import * as ReactDOM from 'react-dom';
-import { Store } from '../../../store/store';
-import { StringProperty } from '../../../store/styleguide/property/string-property';
-
-export interface PreviewAppState {
-	page?: Page;
-}
-
-interface PreviewProps {
-	page?: Page;
-	patternFactories: Map<string, React.StatelessComponent | ObjectConstructor>;
-	selectedElementId?: string;
-}
-
-interface PatternWrapperState {
-	errorMessage?: string;
-}
-
-interface PatternWrapperProps {
-	element: PageElement;
-}
-
-class PatternWrapper extends React.Component<PatternWrapperProps, PatternWrapperState> {
-	public constructor(props: PatternWrapperProps) {
-		super(props);
-		this.state = {};
-	}
-
-	public componentDidCatch(error: Error): void {
-		this.setState({ errorMessage: error.message });
-	}
-
-	public render(): React.ReactNode {
-		if (this.state.errorMessage) {
-			return (
-				<ErrorMessage
-					patternName={this.props.element.getName()}
-					error={this.state.errorMessage}
-				/>
-			);
-		} else {
-			return <div data-sketch-name={this.props.element.getName()}>{this.props.children}</div>;
-		}
-	}
-}
-
-@observer
-class Preview extends React.Component<PreviewProps> {
-	@MobX.observable protected highlightArea: HighlightArea;
-	protected patternWrapperRef: PatternWrapper;
-
-	public constructor(props: PreviewProps) {
-		super(props);
-		this.highlightArea = new HighlightArea();
-	}
-
-	// tslint:disable-next-line:no-any
-	protected collectChildren(componentProps: any, pageElement: PageElement): void {
-		const pattern = pageElement.getPattern();
-
-		if (pattern) {
-			pattern.getSlots().forEach(slot => {
-				const slotId = slot.getId();
-				const reactPropertyName = slotId === Pattern.DEFAULT_SLOT_ID ? 'children' : slotId;
-				const slotContents = pageElement.getSlotContents(slotId);
-				componentProps[reactPropertyName] = slotContents.map((child, index) =>
-					this.createComponent(child)
-				);
-			});
-		}
-	}
-
-	// tslint:disable-next-line:no-any
-	protected collectPropertyValues(componentProps: any, pageElement: PageElement): void {
```

Comment:
This method is missing in the new solution and the reason why enums fail.
All property values must be converted using property.convertToRender.

Reply:
Resolved via 6170e7615862569194e3b8e656a52a265ef90e30

Reply:
Appears to be still broken, could you have another look @TheReincarnator?

Reply:
I tested with the first `<Headline/>` - changing `Level` and `TextAlign` does nothing. Changing the `Size` of a `<Space>`, does though.

Reply:
Scratch that, this is fixed.